train_gpt2.cu

```diff
@@ -1147,11 +1147,10 @@ __device__ float lerp(float start, float end, float weight) {
 }
 
 template <typename Tp, typename Tg>
-__global__ void adamw_kernel3(Tp* params_memory, float* master_params_memory, Tg* grads_memory, float* m_memory, float* v_memory, size_t num_parameters,
-                              float learning_rate, float beta1, float beta2, float beta1_correction, float beta2_correction, float eps, float weight_decay,
+__global__ void adamw_kernel3(Tp* params_memory, float* master_params_memory, Tg* grads_memory, float* m_memory, float* v_memory,
+                              float beta1, float beta2, float learning_rate_inv_beta1_correction, float inv_beta2_correction, float eps, float learning_rate_weight_decay,
                               float grad_scale, unsigned int seed) {
-    int idx = blockIdx.x * blockDim.x + threadIdx.x;
-    if (idx >= num_parameters) { return; }  // guard
```

Comment:
I'd be in favour of keeping the check. it doesn't cost us anything (measurable), and not having it there makes  the kernels brittle. If we change the kernel to vectorized loads, our requirement reg. the number of parameters in the model suddenly may depend on the precision we use for our weights.